functions-app/poison-replay-completed-checks/index.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const fs = require('fs')
+const uuid = require('uuid/v4')
+const path = require('path')
+const compressionService = require('../../functions/lib/compression.service')
+
+module.exports = async function (context, check) {
+  let version
+  let message
+  if (check.version && check.version === '2') {
+    version = 2
+    context.log('decompressing v2 message')
+    message = compressionService.decompress(check.archive)
+  } else {
+    version = 1
+    message = check
+  }
+  const id = uuid()
+  const file = path.join('./', `v${version}`, `${id}.json`)
+  context.log(`saving to ${file}`)
+  try {
+    const jsonString = JSON.stringify(message)
+    fs.writeFileSync(file, jsonString)

```

Comment:
This seems out of place to me in a function.  If we find these messages interesting we should save them somewhere properly like a DB.

For now, surely all this function needs to do is to cycle the poison queue by injecting the messages back onto the processing queue?  Any messages that are not processed will end up back on the poison queue.

Also, why check this function in if it not working?  The write-to-file aspect suggests there might be a lot of additional functionality to come.

Reply:
this function is very much just a sketch / spike to explore peeking messages, reading them and putting back on the queue.  

Why a function?  our azure queue service library in the admin app only includes functionality for adding messages onto a queue, and given current demands i wanted the quickest route to obtain a picture of schools affected.

Ideally this should be in a completely separate PR, as to keep the conversation focused.  So i think ill do just that.